server/src/main/java/org/candlepin/dto/api/v1/PoolDTO.java

```diff
@@ -0,0 +1,1548 @@
+
+        HashCodeBuilder builder = new HashCodeBuilder(37, 7)
+            .append(super.hashCode())
+            .append(this.getId())
+            .append(this.getType())
+            .append(this.getOwner() != null ? this.getOwner().getId() : null)
+            .append(this.isActiveSubscription())
+            .append(this.isCreatedByShare())
+            .append(this.hasSharedAncestor())
+            .append(this.getSourceEntitlement() != null ? this.getSourceEntitlement().getId() : null)
+            .append(this.getQuantity())
+            .append(this.getStartDate())
+            .append(this.getEndDate())
+            .append(this.getAttributes())
+            .append(this.getRestrictedToUsername())
+            .append(this.getContractNumber())
+            .append(this.getAccountNumber())
+            .append(this.getOrderNumber())
+            .append(this.getConsumed())
+            .append(this.getExported())
+            .append(this.getShared())
+            .append(this.getBranding())
+            .append(this.getCalculatedAttributes())
+            .append(this.getUpstreamPoolId())
+            .append(this.getUpstreamEntitlementId())
+            .append(this.getUpstreamConsumerId())
+            .append(this.getProductName())
+            .append(this.getProductId())
+            .append(this.getProductAttributes())
+            .append(this.getStackId())
+            .append(this.isStacked())
+            .append(this.isDevelopmentPool())
+            .append(this.getDerivedProductAttributes())
+            .append(this.getDerivedProductId())
+            .append(this.getDerivedProductName())
+            .append(this.getProvidedProducts())
+            .append(this.getDerivedProvidedProducts())
+            .append(this.getSourceStackId())
+            .append(this.getSubscriptionSubKey())
+            .append(this.getSubscriptionId())
+            .append(this.getCertificate() != null ? this.getCertificate().getId() : null);
+
+        return builder.toHashCode();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PoolDTO clone() {
+        PoolDTO copy = super.clone();
+
+        OwnerDTO owner = this.getOwner();
+        copy.owner = owner != null ? owner.clone() : null;
+
+        CertificateDTO certificate = this.getCertificate();
+        copy.certificate = certificate != null ? certificate.clone() : null;
+
+        copy.setAttributes(this.getAttributes());
+        copy.setCalculatedAttributes(this.getCalculatedAttributes());
+        copy.setProductAttributes(this.getProductAttributes());
+        copy.setDerivedProductAttributes(this.getDerivedProductAttributes());
+        copy.setBranding(this.getBranding());
+        copy.setProvidedProducts(this.getProvidedProducts());
+        copy.setDerivedProvidedProducts(this.getDerivedProvidedProducts());
+
+        copy.endDate = this.endDate != null ? new Date(this.endDate.getTime()) : null;
```

Comment:
Use this.endDate.clone() instead so we're guaranteed to get the same class of date back.